### PR TITLE
feat: improve assets scanning for non-default root

### LINKED ISSUE #201: Should the root of scanning assets be strongly related to `root` in the `vite.config.ts`？
## description

The scanning of assets is now based on the `root` in the `vite.config.ts`. After setting the root to some other path instead of `process.cwd()`, files such as `package.json`, `README.md`, and so on will not display in the assets tab. I think this behavior is a litter strange. We could find a better way to display "all" the asset files in a more natural way.

## repro

URL: <https://stackblitz.com/edit/vitejs-vite-j2b3x7?file=vite.config.ts>

### PR DESCRIPTION
**Pull request by LoTwT**
## description

This PR is to close #201.

### CODE REVIEW

File: packages/core/src/vite-rpc/assets.ts [hunk 1]
```diff
@@ -87,17 +91,22 @@ export function setupAssetsRPC(config: SetupAssetsOptions) {
...
+
       return {
         path,
-        publicPath: join(baseURL, path),
+        publicPath: join(baseURL, publicPath),
         filePath,
         type: guessType(path),
         size: stat.size,
         mtime: stat.mtimeMs,
+        isUnderRoot,
```
Comment (webfansplz) [inline packages/core/src/vite-rpc/assets.ts:109]: => `atRoot`

**Comment by webfansplz** [comment]
image

Seems to have an issue with previewing the public assets.

**Comment by LoTwT** [comment]
> image
> Seems to have an issue with previewing the public assets.

If the asset files are not at the root, they won't be served as static resources by the Vite dev server. I have only hidden the `download` and `code snippets`.

- Should we add some special markers to show that they are not at the root?
- What about adding a new component which may be `AssetPreviewNonRoot` to display them? And we need to discuss how to show them better.

**Comment by alexzhang1030** [comment]
We cannot use `cwd`, cwd may not always point to the codebase root.

For example: 
```
root
  -> packages
      --> front-end
      --> back-end
```
In pnpm workspace (monorepo), we can use commands like `pnpm -F front-end dev`, in this scenario, the `cwd` root, instead of `front-end`, so the scanning result is incorrect.

**Comment by LoTwT** [comment]
Hi @alexzhang1030, after I tried, the `cwd` of `pnpm -F sub-package dev` is `<root>/packages/sub-package` but not the `<root>`, so I think `cwd` might work.

As #201 is not an explicit issue and is outdated, I will close this PR and the issue.

**Pull request closed without merging**